test/distributed/tensor/parallel/test_tp_examples.py

```diff
@@ -215,75 +226,72 @@ def test_transformer_training(self, is_seq_parallel=False):
         if is_seq_parallel:
             assert inp_size[1] % self.world_size == 0
         torch.manual_seed(0)
-        inp = torch.randint(model_args.vocab_size, inp_size, device=self.device_type)
-
-        # Compare outputs on the same input.
-        output = model(inp)
-        with CommDebugMode() as comm_mode:
-            output_tp = model_tp(inp)
-        self.assertEqual(output, output_tp)
-        if is_seq_parallel:
-            self.assertDictEqual(
-                comm_mode.get_comm_counts(),
-                {
-                    c10d_functional.reduce_scatter_tensor: 6,
-                    c10d_functional.all_gather_into_tensor: 6,
-                },
-            )
-        else:
-            self.assertDictEqual(
-                comm_mode.get_comm_counts(),
-                {
-                    c10d_functional.all_reduce: 6,
-                    c10d_functional.all_gather_into_tensor: 1,
-                },
+        steps = 10 if dtype is torch.float64 else 1
```

Comment:
to verify numeric parity for multiple iterations (eg 10), I have to bring back `troch.float64` from previous PR

for float64, I also bring back `module_tp.model_args.n_heads // device_mesh.size()`
for float32, it's still DTensor as input for attention

Reply:
For float8, are you still going to use the math attention?
For fp16 IIRC there is numerical issues with that https://github.com/pytorch/torchtitan/issues/317

A related issue is #125741

Reply:
the related issues are very helpful! didnt know there was stories behind this

for float8 model, sdpa is still in high precision ( thus can be math attention in fp64)

Reply:
I discussed the incompatibility between DTensor and math attention, and the conclusion is we should try modifying math attention code to unblock DTensor.

In the very short term, do we feel we have to test fp8 for math attention? If that's the case, I'm ok with adding back the "hacks" for now.

Reply:
> we should try modifying math attention code to unblock DTensor

that sounds exciting

> In the very short term, do we feel we have to test fp8 for math attention? If that's the case, I'm ok with adding back the "hacks" for now.

Yes, I prefer to have a 10-iteration working now so I can build on top of it. Once DTensor + math attention worked, we can remove the hack around `n_heads // device_mesh.size()`. what do you think ?